src/test/java/uk/nhs/prm/deduction/e2e/tests/RepositoryE2ETests.java

```diff
@@ -112,6 +150,134 @@ void shouldIdentifyEhrRequestAsEhrOutMessage() {
+
+        String smallEhr = Resources.readTestResourceFileFromEhrDirectory("small-ehr-without-linebreaks")
+                .replaceAll("1632CD65-FD8F-4914-B62A-9763B50FC04A", inboundConversationId.toUpperCase())
+                .replaceAll("0206C270-E9A0-11ED-808B-AC162D1F16F0", smallEhrMessageId);
+
+        String ehrRequest = Resources.readTestResourceFile("RCMR_IN010000UK05")
+                .replaceAll("9692842304", nhsNumberForTestPatient)
+                .replaceAll("A91720", sourceGpForTestPatient)
+                .replaceAll("200000000631", "200000000149")
+                .replaceAll("17a757f2-f4d2-444e-a246-9cb77bef7f22", outboundConversationId);
+
+        // When
+        // change transfer db status to ACTION:EHR_REQUEST_SENT before putting on inbound queue
+        // Put the patient to inboundQueueFromMhs as an UK05 message
+
+        trackerDb.save(new TransferTrackerDbMessage(
+                inboundConversationId,
+                "",
+                randomNemsMessageId(),
+                nhsNumberForTestPatient,
+                sourceGpForTestPatient,
+                "ACTION:EHR_REQUEST_SENT",
+                timeNow,
+                timeNow,
+                timeNow
+        ));
+
+        inboundQueueFromMhs.sendMessage(smallEhr, inboundConversationId);
+        LOGGER.info("conversationIdExists: {}",trackerDb.conversationIdExists(inboundConversationId));
+        var status = trackerDb.waitForStatusMatching(inboundConversationId, "ACTION:EHR_TRANSFER_TO_REPO_COMPLETE");
+        LOGGER.info("tracker db status: {}", status);
+
+        // Put a EHR request to inboundQueueFromMhs
+        inboundQueueFromMhs.sendMessage(ehrRequest, outboundConversationId);
+
+        // Then
+        // assert gp2gpMessenger queue got a message of UK06
+        SqsMessage gp2gpMessage = gp2gpMessengerQueue.getMessageContaining(outboundConversationId);
+
+        assertThat(gp2gpMessage).isNotNull();
+        assertThat(gp2gpMessage.contains("RCMR_IN030000UK06")).isTrue();
+
+        try {
+            String gp2gpMessengerPayload = getPayload(gp2gpMessage.body());
+            String smallEhrPayload = getPayload(smallEhr);
+
+            LOGGER.info("Payload from gp2gpMessenger: {}", gp2gpMessengerPayload);
+            LOGGER.info("Payload from smallEhr: {}", smallEhrPayload);
+            Diff myDiff = DiffBuilder.compare(gp2gpMessengerPayload).withTest(smallEhrPayload)
+                    .withNodeMatcher(new DefaultNodeMatcher(ElementSelectors.byName))
+                    .withNodeFilter(this::excludeComparisons)
+                    .withDifferenceEvaluator(DifferenceEvaluators.chain(DifferenceEvaluators.Default,
+                            DifferenceEvaluators.downgradeDifferencesToEqual(ComparisonType.XML_STANDALONE)))
+                    .checkForSimilar().build();
+
+            assertFalse(myDiff.toString(), myDiff.hasDifferences());
+
+        } catch (JSONException e) {
```

Comment:
Small change - completely optional 😄 - could we use 'exception' rather than 'e', it is commonly used and perfectly fine though just for clarification

Reply:
additionally if you do decide to go with this can we reflect it throughout the tests!